crates/proof-of-sql/src/sql/proof_plans/sort_merge_join_exec.rs

```diff
@@ -0,0 +1,600 @@
+        )?;
+        // 7. Membership checks - ordered set union
+        let left_on_col_positions = self
+            .left_join_columns
+            .iter()
+            .map(|col| {
+                self.left_selected_column_ident_aliases
+                    .iter()
+                    .position(|(a, _b)| *a == col.column_id())
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+        let right_on_col_positions = self
+            .right_join_columns
+            .iter()
+            .map(|col| {
+                self.right_selected_column_ident_aliases
+                    .iter()
+                    .position(|(a, _b)| *a == col.column_id())
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+        let left_on_col_evals = left_on_col_positions
+            .iter()
+            .map(|&pos| left_col_evals[pos])
+            .collect::<Vec<_>>();
+        let right_on_col_evals = right_on_col_positions
+            .iter()
+            .map(|&pos| right_col_evals[pos])
+            .collect::<Vec<_>>();
+        if left_col_evals.len() > 1 {
+            return Err(ProofError::VerificationError {
+                error: "Join on multiple columns not supported yet",
+            });
+        }
+        let res_left_on_col_evals = left_on_col_positions
+            .iter()
+            .map(|&pos| res_col_evals[pos])
+            .collect::<Vec<_>>();
+        let res_right_on_col_evals = right_on_col_positions
+            .iter()
+            .map(|&pos| res_col_evals[pos + num_columns_left])
+            .collect::<Vec<_>>();
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            g_one_eval,
+            left_one_eval,
+            &[g_col_eval],
+            &left_on_col_evals,
+        )?;
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            g_one_eval,
+            right_one_eval,
+            &[g_col_eval],
+            &right_on_col_evals,
+        )?;
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            g_one_eval,
+            res_one_eval,
+            &[g_col_eval],
+            &res_left_on_col_evals,
+        )?;
+        dbg!("7 steps done");
+        // 8. Join Condition Check: res_left_on = res_right_on
+        for (left, right) in res_left_on_col_evals.iter().zip_eq(res_right_on_col_evals) {
+            if *left != right {
+                return Err(ProofError::VerificationError {
+                    error: "Join condition check failed",
+                });
+            }
+        }
+        // 9. Monotonicity checks
+        verify_monotonic::<S, true, true>(builder, alpha, beta, g_col_eval, g_one_eval)?;
+        verify_monotonic::<S, true, true>(
+            builder,
+            alpha,
+            beta,
+            sorted_combined_indexes_eval,
```

Comment:
We need to add a check that `sorted_combined_indexes_eval` is correct.

sorted_combined_indexes_eval = 2^64 * res_left_index + res_right_index